src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -17,11 +18,16 @@ namespace System.Net.Security.Tests
     public class SslStreamNetworkStreamTest
     {
         private readonly X509Certificate2 _serverCert;
-        private readonly X509CertificateCollection _serverChain;
+        private readonly X509Certificate2Collection _serverChain;
 
         public SslStreamNetworkStreamTest()
         {
-            (_serverCert, _serverChain) = TestHelper.GenerateCertificates("localhost");
+            (_serverCert, _serverChain) = TestHelper.GenerateCertificates("localhost", this.GetType().Name);
+        }
+
+        ~SslStreamNetworkStreamTest()
```

Comment:
Perhaps `Dispose` instead? If you need clean up after test has finished then XUnit recognizes `IDisposable` and uses it as a clean up method: https://xunit.net/docs/shared-context#constructor